pkg/controller/seed-controller-manager/mla/resources.go

```diff
@@ -397,3 +451,84 @@ func getOrgByProjectID(ctx context.Context, client ctrlruntimeclient.Client, gra
 	}
 	return grafanaClient.GetOrgById(ctx, uint(id))
 }
+
+// GatewayCACreator returns a function to create the ECDSA-based CA to be used for MLA Gateway.
+func GatewayCACreator() reconciling.NamedSecretCreatorGetter {
+	return func() (string, reconciling.SecretCreator) {
+		return resources.MLAGatewayCASecretName, func(se *corev1.Secret) (*corev1.Secret, error) {
+			if se.Data == nil {
+				se.Data = map[string][]byte{}
+			}
+
+			if data, exists := se.Data[resources.MLAGatewayCACertKey]; exists {
+				certs, err := certutil.ParseCertsPEM(data)
+				if err != nil {
+					return nil, fmt.Errorf("failed to parse certificate %s from existing secret %s: %v",
+						resources.MLAGatewayCACertKey, resources.MLAGatewayCASecretName, err)
+				}
+				if !resources.CertWillExpireSoon(certs[0]) {
+					return se, nil
+				}
+			}
+
+			cert, key, err := certificates.GetECDSACACertAndKey()
+			if err != nil {
+				return nil, fmt.Errorf("failed to generate MLA CA: %v", err)
+			}
+			se.Data[resources.MLAGatewayCACertKey] = cert
+			se.Data[resources.MLAGatewayCAKeyKey] = key
+
+			return se, nil
+		}
+	}
+}
+
+type tlsServerCertCreatorData interface {
+	GetMLAGatewayCA() (*resources.ECDSAKeyPair, error)
+}
+
+// GatewayCertificateCreator returns a function to create/update a secret with the MLA gateway TLS certificate.
+func GatewayCertificateCreator(data tlsServerCertCreatorData, c *kubermaticv1.Cluster) reconciling.NamedSecretCreatorGetter {
```

Comment:
Instead of creating a new interface which is private and will be only used here, I would inline this interfaces method like this: 
```suggestion
func GatewayCertificateCreator(fn func() (*resources.ECDSAKeyPair, error), c *kubermaticv1.Cluster) reconciling.NamedSecretCreatorGetter {
```

And in the line 498 would call that function. This way you keep the scope of this abstraction bounded to the corresponding method instead of introducing a new packaged-scope interface that is used in one position.